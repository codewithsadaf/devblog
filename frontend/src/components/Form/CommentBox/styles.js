//imports
import styled, { css } from "styled-components";
//components styles
import { InputComponent } from "../Input/styles";
import { ButtonContainer as Button } from "../../Layout/Button/styles";

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & ${Button} {
        padding: 12px 16px;
    }
`;

export const CommentInput = styled(InputComponent)`
    ${({ theme }) => css`
        border: 1px solid ${theme.colors.gray};
        transition: ${theme.transition.tran_03};
        padding: 12px 7px;
        width: 90%;
        border-radius: 5px;
        font-size: 1em;
        margin-right: 7px;

        &:focus {
            border: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
