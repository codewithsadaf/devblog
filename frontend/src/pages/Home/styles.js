//imports
import styled, { css } from "styled-components";

export const HomeContainer = styled.div`
    ${({ theme }) => css`
        width: 600px;
        margin: 3em auto 0;

        @media ${theme.media.lteMedium} {
            max-width: 100%;
        }
    `}
`;

export const MessageWrapper = styled.div`
    margin-top: 5em;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
`;
