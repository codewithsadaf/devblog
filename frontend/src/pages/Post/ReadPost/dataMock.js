export const postMock = {
    _id: "1",
    image: "example.png",
    title: "post title 1",
    content: "<p>post content</p>",
    user: {
        name: "john doe",
    },
    tags: ["#tag1", "#tag2"],
    createdAt: "2023-04-05T19:27:03.125+00:00",
    updatedAt: "2023-04-05T19:27:03.125+00:00",
};

export const commentsMock = [
    {
        user: {
            image: "example.png",
            name: "john doe",
        },
        createdAt: "2023-04-05T19:27:03.125+00:00",
        _id: "1",
        commentText: "this is a comment 01",
    },
    {
        user: {
            image: "example.png",
            name: "jane doe",
            _id: "2",
        },
        createdAt: "2023-04-05T19:27:03.125+00:00",
        _id: "2",
        commentText: "this is a comment 02",
    },
];

export const currentUserMock = {
    name: "john doe",
    _id: "1",
};
