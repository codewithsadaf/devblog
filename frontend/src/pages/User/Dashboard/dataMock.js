export const postsMock = [
    {
        _id: "1",
        image: "example.png",
        title: "post title 1",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag1", "#tag2"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "2",
        image: "example.png",
        title: "post title 2",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag3", "#tag4"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "3",
        image: "example.png",
        title: "post title 3",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag1", "#tag3"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
];

export const commentsMock = [
    {
        user: {
            image: "example.png",
            name: "john doe",
        },
        post: {
            _id: "1",
        },
        createdAt: "2023-04-05T19:27:03.125+00:00",
        _id: "1",
        commentText: "this is a comment 01",
    },
    {
        user: {
            image: "example.png",
            name: "john doe",
            _id: "2",
        },
        post: {
            _id: "2",
        },
        createdAt: "2023-04-05T19:27:03.125+00:00",
        _id: "2",
        commentText: "this is a comment 02",
    },
];

export const currentUserMock = {
    currentUser: {
        name: "john doe",
        image: "image.png",
        email: "john@test.com",
        createdAt: "2023-04-05T19:27:03.125+00:00",
    },
};
