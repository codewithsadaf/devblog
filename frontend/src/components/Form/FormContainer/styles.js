//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Form = styled.form`
    ${({ theme }) => css`
        width: 450px;
        margin-top: 3em;

        & > ${Button} {
            margin-top: 1em;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
            min-width: 0;
            max-width: 100%;
        }
    `}
`;
