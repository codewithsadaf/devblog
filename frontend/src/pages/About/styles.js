//imports
import styled, { css } from "styled-components";
//components styles
import { Text as TextComponent } from "../../components/Layout/TextComponent/styles";
import { Container as Heading } from "../../components/Layout/Heading/styles";

export const AboutContainer = styled.div`
    ${({ theme }) => css`
        max-width: 60%;
        margin: 3em auto 0;
        text-align: center;

        & > ${TextComponent} {
            margin-top: 0.5em;
        }

        @media ${theme.media.lteMedium} {
            max-width: 100%;

            & > ${Heading} {
                font-size: ${theme.font.sizes.xxlarge};
                line-height: ${theme.font.line_height.h3};
            }
        }
    `}
`;
