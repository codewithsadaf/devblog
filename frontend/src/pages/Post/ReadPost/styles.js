//imports
import styled, { css } from "styled-components";
import { Link } from "react-router-dom";
//components styles
import { ImageElm } from "../../../components/Layout/ImageBox/styles";
import { Text as TextComponent } from "../../../components/Layout/TextComponent/styles";

//PAGE CONTAINER
export const Container = styled.div`
    margin: 3em auto 0;
`;

//RETURN TO HOME TEXT
export const BackWrapper = styled.div`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin-bottom: 1.2em;
    align-items: center;
`;

export const BackLink = styled(Link)`
    ${({ theme }) => css`
        color: ${theme.colors.primaryColor};
        transition: ${theme.transition.tran_03};
        display: flex;
        align-items: center;

        &:hover {
            transition: ${theme.transition.tran_03};
            opacity: 0.7;
        }
    `}
`;

//POST HEADER
export const PostHeader = styled.div`
    ${({ theme }) => css`
        width: 80%;
        margin: 0 auto;
        text-align: center;

        ${ImageElm} {
            height: 50%;
            width: 50%;
            margin: 2em auto;

            @media ${theme.media.lteMedium} {
                width: 100%;
                height: 100%;
            }
        }

        ${TextComponent} {
            margin: 1em 0;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

//POST CONTENT
export const ContentWrapper = styled.div`
    ${({ theme }) => css`
        width: 60%;
        margin: 0 auto;

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const DisplayContent = styled.div`
    ${({ theme }) => css`
        text-align: left;

        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin-bottom: 0.5em;
            font-weight: bold;
        }

        p {
            font-size: 18px;
            margin-bottom: 1em;
        }

        a {
            font-weight: bold;
            color: ${theme.colors.primaryColor};
            text-decoration: underline;
        }

        ol,
        ul {
            margin-left: 2em;

            li {
                margin-bottom: 1em;
            }
        }

        ul {
            list-style-type: disc;
        }

        ol {
            list-style-type: decimal;
        }

        img {
            width: 100%;
            display: block;
            margin: 2em 0;
        }
    `}
`;

//POST TAGS
export const TagsWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    margin: 3em 0;
    flex-wrap: wrap;
`;

export const TagText = styled.span`
    display: block;
    padding: 0 0.5em;
    font-weight: 500;

    &::before {
        content: "#";
        font-weight: bold;
    }
`;

export const TagsInfo = styled.span`
    display: block;
    margin-right: 0.5em;
    font-weight: bold;
`;

//POST LINE
export const Line = styled.div`
    ${({ theme }) => css`
        height: 2px;
        width: 100%;
        background-color: ${theme.colors.footerBackground};
        margin: 1em 0;
    `}
`;

//POST COMMENTS
export const CommentsNumberText = styled.h5`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.large};
        line-height: ${theme.font.line_height.h5};
    `}
`;

export const CommentsWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
`;

export const SignUpWrapper = styled.div`
    margin: 2em 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    width: 100%;
    text-align: center;

    button {
        margin-top: 1em;
    }
`;

export const SignUpText = styled.h4`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    `}
`;
