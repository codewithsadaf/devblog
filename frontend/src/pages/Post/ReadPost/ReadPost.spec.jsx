//imports
import { screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Router from "react-router-dom";
//components
import { ReadPost } from "./";
//styles
import { renderTheme } from "../../../styles/render-theme";
//contexts
import { Context as UserContext } from "../../../context/UserContext";
import { PostContext } from "../../../context/PostContext";
import { CommentContext } from "../../../context/CommentContext";
//mock data
import { commentsMock, currentUserMock, postMock } from "./dataMock";

//mock useParams
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
}));

//spy on useParams
jest.spyOn(Router, "useParams").mockReturnValue({ id: "1234" });

//mock scrollTop
Object.defineProperty(window.global, "scrollTo", { value: jest.fn() });

describe("<ReadPost />", () => {
    let mockCheckUser;
    let mockGetPostById;
    let mockCreateComment;
    let mockEditComment;
    let mockRemoveComment;
    let mockGetCommentsByPostId;

    beforeEach(() => {
        mockCheckUser = jest.fn();
        mockGetPostById = jest.fn();
        mockCreateComment = jest.fn();
        mockEditComment = jest.fn();
        mockRemoveComment = jest.fn();
        mockGetCommentsByPostId = jest.fn();
    });

    describe("Rendering", () => {
        it("should render ReadPost", async () => {
            mockGetPostById.mockReturnValueOnce(Promise.resolve(postMock));
            mockCheckUser.mockReturnValueOnce(Promise.resolve(null));
            mockGetCommentsByPostId.mockReturnValueOnce(
                Promise.resolve(commentsMock)
            );

            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: false,
                            checkUser: mockCheckUser,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                getPostById: mockGetPostById,
                            }}
                        >
                            <CommentContext.Provider
                                value={{
                                    createComment: mockCreateComment,
                                    editComment: mockEditComment,
                                    removeComment: mockRemoveComment,
                                    getCommentsByPostId:
                                        mockGetCommentsByPostId,
                                }}
                            >
                                <ReadPost />
                            </CommentContext.Provider>
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
                expect(mockGetCommentsByPostId).toHaveBeenCalledTimes(1);
                expect(mockCheckUser).toHaveBeenCalledTimes(1);
            });

            const postHeading = screen.getByRole("heading", {
                name: postMock.title,
            });
            const authorText = screen.getByText(
                /Por:\s*john doe\s*-\s*5 de Abril 2023/
            );
            const postImage = screen.getByRole("img", { name: postMock.title });
            const postContent = screen.getByText(
                postMock.content.replace("<p>", "").replace("</p>", "")
            );
            const SignUpButton = screen.getByText(
                "Crie uma conta para comentar."
            );
            const postTagText1 = screen.getByText("tag1");
            const postTagText2 = screen.getByText("tag2");
            const commentText1 = screen.getByText("this is a comment 01");
            const commentText2 = screen.getByText("this is a comment 01");

            expect(postHeading).toBeInTheDocument();
            expect(authorText).toBeInTheDocument();
            expect(postImage).toBeInTheDocument();
            expect(postImage).toHaveAttribute("src", postMock.image);
            expect(postContent).toBeInTheDocument();
            expect(postTagText1).toBeInTheDocument();
            expect(postTagText2).toBeInTheDocument();
            expect(SignUpButton).toBeInTheDocument();
            expect(commentText1).toBeInTheDocument();
            expect(commentText2).toBeInTheDocument();
        });
        it("should render comment input when user is authenticated", async () => {
            mockGetPostById.mockReturnValueOnce(Promise.resolve(postMock));
            mockCheckUser.mockReturnValueOnce(Promise.resolve(currentUserMock));
            mockGetCommentsByPostId.mockReturnValueOnce(
                Promise.resolve(commentsMock)
            );

            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: true,
                            checkUser: mockCheckUser,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                getPostById: mockGetPostById,
                            }}
                        >
                            <CommentContext.Provider
                                value={{
                                    createComment: mockCreateComment,
                                    editComment: mockEditComment,
                                    removeComment: mockRemoveComment,
                                    getCommentsByPostId:
                                        mockGetCommentsByPostId,
                                }}
                            >
                                <ReadPost />
                            </CommentContext.Provider>
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
                expect(mockGetCommentsByPostId).toHaveBeenCalledTimes(1);
                expect(mockCheckUser).toHaveBeenCalledTimes(1);
            });

            const commentInput = screen.getByPlaceholderText(
                "Deixe seu comentÃ¡rio aqui..."
            );

            expect(commentInput).toBeInTheDocument();
        });
        it("should render comment input with different placeholder Text", async () => {
            mockGetPostById.mockReturnValueOnce(Promise.resolve(postMock));
            mockCheckUser.mockReturnValueOnce(Promise.resolve(currentUserMock));
            mockGetCommentsByPostId.mockReturnValueOnce(Promise.resolve([]));

            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: true,
                            checkUser: mockCheckUser,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                getPostById: mockGetPostById,
                            }}
                        >
                            <CommentContext.Provider
                                value={{
                                    createComment: mockCreateComment,
                                    editComment: mockEditComment,
                                    removeComment: mockRemoveComment,
                                    getCommentsByPostId:
                                        mockGetCommentsByPostId,
                                }}
                            >
                                <ReadPost />
                            </CommentContext.Provider>
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
                expect(mockGetCommentsByPostId).toHaveBeenCalledTimes(1);
                expect(mockCheckUser).toHaveBeenCalledTimes(1);
            });

            const commentInput = screen.getByPlaceholderText(
                "Seja o primeiro a comentar!"
            );

            expect(commentInput).toBeInTheDocument();
        });
        it("should not render pege if post data fails to fetch", async () => {
            mockGetPostById.mockReturnValueOnce(Promise.resolve(null));
            mockCheckUser.mockReturnValueOnce(Promise.resolve(currentUserMock));
            mockGetCommentsByPostId.mockReturnValueOnce(Promise.resolve([]));

            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: true,
                            checkUser: mockCheckUser,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                getPostById: mockGetPostById,
                            }}
                        >
                            <CommentContext.Provider
                                value={{
                                    createComment: mockCreateComment,
                                    editComment: mockEditComment,
                                    removeComment: mockRemoveComment,
                                    getCommentsByPostId:
                                        mockGetCommentsByPostId,
                                }}
                            >
                                <ReadPost />
                            </CommentContext.Provider>
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
                expect(mockGetCommentsByPostId).toHaveBeenCalledTimes(1);
                expect(mockCheckUser).toHaveBeenCalledTimes(1);
            });

            const postHeading = screen.queryByRole("heading", {
                name: postMock.title,
            });
            const authorText = screen.queryByText(
                /Por:\s*john doe\s*-\s*5 de Abril 2023/
            );
            const postImage = screen.queryByRole("img", {
                name: postMock.title,
            });
            const postContent = screen.queryByText(
                postMock.content.replace("<p>", "").replace("</p>", "")
            );
            const SignUpButton = screen.queryByText(
                "Crie uma conta para comentar."
            );
            const postTagText1 = screen.queryByText("tag1");
            const postTagText2 = screen.queryByText("tag2");
            const commentText1 = screen.queryByText("this is a comment 01");
            const commentText2 = screen.queryByText("this is a comment 01");

            expect(postHeading).not.toBeInTheDocument();
            expect(authorText).not.toBeInTheDocument();
            expect(postImage).not.toBeInTheDocument();
            expect(postContent).not.toBeInTheDocument();
            expect(postTagText1).not.toBeInTheDocument();
            expect(postTagText2).not.toBeInTheDocument();
            expect(SignUpButton).not.toBeInTheDocument();
            expect(commentText1).not.toBeInTheDocument();
            expect(commentText2).not.toBeInTheDocument();
        });
    });
    it("should match snapshot", async () => {
        mockGetPostById.mockReturnValueOnce(Promise.resolve(postMock));
        mockCheckUser.mockReturnValueOnce(Promise.resolve(null));
        mockGetCommentsByPostId.mockReturnValueOnce(
            Promise.resolve(commentsMock)
        );

        const { container } = renderTheme(
            <BrowserRouter>
                <UserContext.Provider
                    value={{
                        authenticated: false,
                        checkUser: mockCheckUser,
                    }}
                >
                    <PostContext.Provider
                        value={{
                            getPostById: mockGetPostById,
                        }}
                    >
                        <CommentContext.Provider
                            value={{
                                createComment: mockCreateComment,
                                editComment: mockEditComment,
                                removeComment: mockRemoveComment,
                                getCommentsByPostId: mockGetCommentsByPostId,
                            }}
                        >
                            <ReadPost />
                        </CommentContext.Provider>
                    </PostContext.Provider>
                </UserContext.Provider>
            </BrowserRouter>
        );

        await waitFor(() => {
            expect(mockGetPostById).toHaveBeenCalledTimes(1);
            expect(mockGetCommentsByPostId).toHaveBeenCalledTimes(1);
            expect(mockCheckUser).toHaveBeenCalledTimes(1);
        });

        expect(container).toMatchSnapshot();
    });
});
