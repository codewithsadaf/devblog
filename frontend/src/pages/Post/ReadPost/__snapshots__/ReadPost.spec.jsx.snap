// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadPost /> should match snapshot 1`] = `
.c8 {
  width: 150px;
  height: 150px;
  object-fit: cover;
  display: block;
  margin: 1em auto;
  border-radius: 7px;
}

.c6 {
  font-size: 1em;
  line-height: 150%;
}

.c0 {
  margin: 3em auto 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 1.2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:hover {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  opacity: 0.7;
}

.c1 {
  width: 80%;
  margin: 0 auto;
  text-align: center;
}

.c1 .c7 {
  height: 50%;
  width: 50%;
  margin: 2em auto;
}

.c1 .c5 {
  margin: 1em 0;
}

.c9 {
  width: 60%;
  margin: 0 auto;
}

.c10 {
  text-align: left;
}

.c10 h1,
.c10 h2,
.c10 h3,
.c10 h4,
.c10 h5,
.c10 h6 {
  margin-bottom: 0.5em;
  font-weight: bold;
}

.c10 p {
  font-size: 18px;
  margin-bottom: 1em;
}

.c10 a {
  font-weight: bold;
  color: #000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10 ol,
.c10 ul {
  margin-left: 2em;
}

.c10 ol li,
.c10 ul li {
  margin-bottom: 1em;
}

.c10 ul {
  list-style-type: disc;
}

.c10 ol {
  list-style-type: decimal;
}

.c10 img {
  width: 100%;
  display: block;
  margin: 2em 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: 3em 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c13 {
  display: block;
  padding: 0 0.5em;
  font-weight: 500;
}

.c13::before {
  content: "#";
  font-weight: bold;
}

.c12 {
  display: block;
  margin-right: 0.5em;
  font-weight: bold;
}

.c15 {
  height: 2px;
  width: 100%;
  background-color: #edf3f6;
  margin: 1em 0;
}

.c14 {
  font-size: 1.4em;
  line-height: 120%;
}

.c19 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  margin: 2em 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  text-align: center;
}

.c16 button {
  margin-top: 1em;
}

.c17 {
  font-size: 1.7em;
  line-height: 120%;
}

.c4 {
  text-transform: none;
  font-weight: bold;
  font-size: 2.9em;
  line-height: 110%;
}

.c23 {
  text-transform: none;
  font-weight: bold;
  font-size: 1.2em;
  line-height: 120%;
}

.c24 {
  text-transform: none;
  font-weight: 500;
  font-size: 0.8em;
  line-height: 150%;
}

.c18 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: #000;
  color: #fff;
  border: 2px solid #000;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c18:hover {
  background: transparent;
  color: #000;
  border: 2px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c20 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 2em 0;
  padding: 0 7px;
  position: relative;
}

.c21 {
  min-width: 7%;
  width: 7%;
  height: 100%;
  margin: 0 1em 0 0;
}

.c21 .c7 {
  width: 100%;
  height: 100%;
  margin: 0;
}

.c22 {
  width: 93%;
  border-bottom: 1px solid #ccc;
  padding-bottom: 1em;
}

.c22 > .c5 {
  margin-top: 1.2em;
  display: block;
  width: 100%;
  word-wrap: break-word;
}

.c25 {
  position: absolute;
  right: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c27 {
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  font-size: 1em;
}

.c27:hover {
  color: #a40404;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c26 {
  cursor: pointer;
  margin-right: 0.5em;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  font-size: 1em;
}

.c26:hover {
  color: #04a453;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

@media (max-width:768px) {
  .c1 .c7 {
    width: 100%;
    height: 100%;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c9 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 1.7em;
    line-height: 120%;
  }
}

@media (max-width:768px) {
  .c21 {
    width: 10%;
    min-width: 10%;
  }
}

@media (max-width:768px) {
  .c22 {
    width: 90%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          Voltar
        </a>
      </div>
      <h1
        class="c4"
      >
        post title 1
      </h1>
      <p
        class="c5 c6"
      >
        Por: 
        john doe
         - 
        5 de Abril 2023
      </p>
      <div
        class=""
      >
        <img
          alt="post title 1"
          class="c7 c8"
          src="example.png"
        />
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="c10"
      >
        <p>
          post content
        </p>
      </div>
      <div
        class="c11"
      >
        <span
          class="c12"
        >
          Tags: 
        </span>
        <span
          class="c13"
        >
          tag1
        </span>
        <span
          class="c13"
        >
          tag2
        </span>
      </div>
      <h5
        class="c14"
      >
        Coment√°rios (2)
      </h5>
      <div
        class="c15"
      />
      <div
        class="c16"
      >
        <h4
          class="c17"
        >
          Crie uma conta para comentar.
        </h4>
        <button
          class="c18"
          type="button"
        >
          Criar uma conta
        </button>
      </div>
      <div
        class="c19"
      >
        <div
          class="c20"
        >
          <div
            class="c21"
          >
            <div
              class=""
            >
              <img
                alt="User Image"
                class="c7 c8"
                src="example.png"
              />
            </div>
          </div>
          <div
            class="c22"
          >
            <h6
              class="c23"
            >
              john doe
            </h6>
            <h1
              class="c24"
            >
              5 de Abril 2023
            </h1>
            <p
              class="c5 c6"
            >
              this is a comment 01
            </p>
          </div>
          <div
            class="c25"
          >
            <svg
              aria-label="Edit Icon"
              class="c26"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
            <svg
              aria-label="Remove Icon"
              class="c27"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              />
            </svg>
          </div>
        </div>
        <div
          class="c20"
        >
          <div
            class="c21"
          >
            <div
              class=""
            >
              <img
                alt="User Image"
                class="c7 c8"
                src="example.png"
              />
            </div>
          </div>
          <div
            class="c22"
          >
            <h6
              class="c23"
            >
              jane doe
            </h6>
            <h1
              class="c24"
            >
              5 de Abril 2023
            </h1>
            <p
              class="c5 c6"
            >
              this is a comment 02
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
